Fix import types from `debug`

- Markuplint version: `3.8.0`
- TypeScript version: `5.0.4`
- Node.js version: `v18.16.0`
- OS: Ubuntu 22.04.2 LTS

## Describe the bug

Markuplint imports types from [`debug`](https://www.npmjs.com/package/debug), but this package doesn't export types. The types are in the [`@types/debug`](https://www.npmjs.com/package/@types/debug) package.

[`@markuplint/ml-core@3.8.0/lib/debug.d.ts`](https://cdn.jsdelivr.net/npm/@markuplint/ml-core@3.8.0/lib/debug.d.ts):

  ```TypeScript
  import debug from 'debug';
  export declare const log: debug.Debugger;
  export declare function enableDebug(): void;
  ```

## Code Example

- `package.json`:

  ```JSON
  {
      "name": "testcase",
      "version": "1.0.0",
      "private": true,
      "dependencies": {
          "markuplint": "3.8.0",
          "typescript": "5.0.4"
      }
  }
  ```

- `index.ts`:

  ```TypeScript
  import { MLEngine } from "markuplint";

  console.log(MLEngine);
  ```

## Steps To Reproduce

1. `npm install`
2. `npx tsc --lib esnext,dom --target esnext --moduleResolution nodenext --strict index.ts`

## The current behavior

```
node_modules/@markuplint/ml-core/lib/debug.d.ts:1:19 - error TS7016: Could not find a declaration file for module 'debug'. '/home/regseb/testcase/node_modules/debug/src/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/debug` if it exists or add a new declaration (.d.ts) file containing `declare module 'debug';`

1 import debug from 'debug';
                    ~~~~~~~

node_modules/strict-event-emitter/lib/StrictEventEmitter.d.ts:1:30 - error TS7016: Could not find a declaration file for module 'events'. '/home/regseb/testcase/node_modules/events/events.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/events` if it exists or add a new declaration (.d.ts) file containing `declare module 'events';`

1 import { EventEmitter } from 'events';
                               ~~~~~~~~


Found 2 errors in 2 files.

Errors  Files
     1  node_modules/@markuplint/ml-core/lib/debug.d.ts:1
     1  node_modules/strict-event-emitter/lib/StrictEventEmitter.d.ts:1
```

## The expected behavior

No errors (or only the `'events'` one).

--- a/packages/@markuplint/astro-parser/package.json
+++ b/packages/@markuplint/astro-parser/package.json
@@ -15,9 +15,6 @@
 		"build": "tsc",
 		"clean": "tsc --build --clean"
 	},
-	"devDependencies": {
-		"@astrojs/parser": "0.22.2"
-	},
 	"dependencies": {
 		"@astrojs/parser": "0.22.2",
 		"@markuplint/html-parser": "3.6.1",

--- a/packages/@markuplint/file-resolver/package.json
+++ b/packages/@markuplint/file-resolver/package.json
@@ -29,7 +29,6 @@
 		"@markuplint/ml-spec": "3.7.0",
 		"@markuplint/shared": "3.5.0",
 		"@types/cosmiconfig": "^6.0.0",
-		"@types/glob": "^8.0.0",
 		"cosmiconfig": "^8.0.0",
 		"cosmiconfig-typescript-loader": "^4.3.0",
 		"glob": "^10.2.2",

--- a/packages/@markuplint/file-resolver/package.json
+++ b/packages/@markuplint/file-resolver/package.json
@@ -19,8 +19,6 @@
 		"clean": "tsc --build --clean"
 	},
 	"devDependencies": {
-		"@types/cosmiconfig": "^6.0.0",
-		"@types/glob": "^8.0.0",
 		"@types/node": "18.16.3"
 	},
 	"dependencies": {
@@ -30,6 +28,8 @@
 		"@markuplint/ml-core": "3.8.0",
 		"@markuplint/ml-spec": "3.7.0",
 		"@markuplint/shared": "3.5.0",
+		"@types/cosmiconfig": "^6.0.0",
+		"@types/glob": "^8.0.0",
 		"cosmiconfig": "^8.0.0",
 		"cosmiconfig-typescript-loader": "^4.3.0",
 		"glob": "^10.2.2",

--- a/packages/@markuplint/html-parser/package.json
+++ b/packages/@markuplint/html-parser/package.json
@@ -23,9 +23,7 @@
 		"@markuplint/ml-ast": "3.1.0",
 		"@markuplint/parser-utils": "3.6.1",
 		"parse5": "7.1.2",
-		"tslib": "^2.4.1"
-	},
-	"devDependencies": {
+		"tslib": "^2.4.1",
 		"type-fest": "^3.8.0"
 	}
 }

--- a/packages/@markuplint/ml-config/package.json
+++ b/packages/@markuplint/ml-config/package.json
@@ -18,13 +18,11 @@
 		"build": "tsc",
 		"clean": "tsc --build --clean"
 	},
-	"devDependencies": {
-		"@markuplint/ml-ast": "3.1.0",
-		"@markuplint/shared": "3.5.0",
-		"@types/mustache": "^4.2.2"
-	},
 	"dependencies": {
+		"@markuplint/ml-ast": "3.1.0",
 		"@markuplint/selector": "3.7.0",
+		"@markuplint/shared": "3.5.0",
+		"@types/mustache": "^4.2.2",
 		"deepmerge": "^4.3.1",
 		"is-plain-object": "^5.0.0",
 		"mustache": "^4.2.0",

--- a/packages/@markuplint/ml-core/package.json
+++ b/packages/@markuplint/ml-core/package.json
@@ -32,12 +32,10 @@
 		"@markuplint/ml-spec": "3.7.0",
 		"@markuplint/parser-utils": "3.6.1",
 		"@markuplint/selector": "3.7.0",
+		"@types/debug": "^4.1.7",
 		"debug": "^4.3.4",
 		"is-plain-object": "^5.0.0",
-		"tslib": "^2.4.1"
-	},
-	"devDependencies": {
-		"@types/debug": "^4.1.7",
+		"tslib": "^2.4.1",
 		"type-fest": "^3.8.0"
 	}
 }

--- a/packages/@markuplint/ml-spec/package.json
+++ b/packages/@markuplint/ml-spec/package.json
@@ -26,14 +26,14 @@
 	},
 	"dependencies": {
 		"@markuplint/ml-ast": "3.1.0",
+		"@markuplint/types": "3.5.1",
 		"dom-accessibility-api": "^0.5.14",
 		"is-plain-object": "^5.0.0",
-		"tslib": "^2.4.1"
+		"tslib": "^2.4.1",
+		"type-fest": "^3.8.0"
 	},
 	"devDependencies": {
 		"@markuplint/test-tools": "3.2.0",
-		"@markuplint/types": "3.5.1",
-		"json-schema-to-typescript": "11.0.2",
-		"type-fest": "^3.8.0"
+		"json-schema-to-typescript": "11.0.2"
 	}
 }

--- a/packages/@markuplint/parser-utils/package.json
+++ b/packages/@markuplint/parser-utils/package.json
@@ -18,14 +18,12 @@
 		"build": "tsc",
 		"clean": "tsc --build --clean"
 	},
-	"devDependencies": {
-		"@types/uuid": "^9.0.0",
-		"type-fest": "^3.8.0"
-	},
 	"dependencies": {
 		"@markuplint/ml-ast": "3.1.0",
 		"@markuplint/types": "3.5.1",
+		"@types/uuid": "^9.0.0",
 		"tslib": "^2.4.1",
+		"type-fest": "^3.8.0",
 		"uuid": "^9.0.0"
 	},
 	"peerDependencies": {

--- a/packages/@markuplint/react-spec/package.json
+++ b/packages/@markuplint/react-spec/package.json
@@ -15,7 +15,7 @@
 		"build": "tsc",
 		"clean": "tsc --build --clean"
 	},
-	"devDependencies": {
+	"dependencies": {
 		"@markuplint/ml-spec": "3.7.0"
 	},
 	"peerDependencies": {

--- a/packages/@markuplint/rules/package.json
+++ b/packages/@markuplint/rules/package.json
@@ -26,14 +26,12 @@
 		"@markuplint/selector": "3.7.0",
 		"@markuplint/shared": "3.5.0",
 		"@markuplint/types": "3.5.1",
+		"@types/debug": "^4.1.7",
 		"@ungap/structured-clone": "^1.0.1",
 		"ansi-colors": "^4.1.3",
 		"chrono-node": "^2.5.0",
 		"debug": "^4.3.4",
-		"tslib": "^2.4.1"
-	},
-	"devDependencies": {
-		"@types/debug": "^4.1.7",
+		"tslib": "^2.4.1",
 		"type-fest": "^3.8.0"
 	}
 }

--- a/packages/@markuplint/selector/package.json
+++ b/packages/@markuplint/selector/package.json
@@ -21,14 +21,14 @@
 	},
 	"dependencies": {
 		"@markuplint/ml-spec": "3.7.0",
+		"@types/debug": "^4.1.7",
 		"debug": "^4.3.4",
 		"postcss-selector-parser": "^6.0.11",
-		"tslib": "^2.4.1"
+		"tslib": "^2.4.1",
+		"type-fest": "^3.8.0"
 	},
 	"devDependencies": {
-		"@types/debug": "^4.1.7",
 		"@types/jsdom": "21.1.1",
-		"jsdom": "21.1.1",
-		"type-fest": "^3.8.0"
+		"jsdom": "21.1.1"
 	}
 }

--- a/packages/@markuplint/types/package.json
+++ b/packages/@markuplint/types/package.json
@@ -20,17 +20,15 @@
 		"schema": "ts-node './gen/types.ts'; json2ts './types.schema.json' > './src/types.schema.ts'; prettier './src/types.schema.ts' './types.schema.json' --write; eslint './src/types.schema.ts' --fix; tsc;"
 	},
 	"dependencies": {
-		"bcp-47": "1",
-		"css-tree": "^2.3.1",
-		"debug": "^4.3.4",
-		"leven": "3",
-		"whatwg-mimetype": "^3.0.0"
-	},
-	"devDependencies": {
 		"@types/bcp-47": "1",
 		"@types/css-tree": "^2.0.1",
 		"@types/debug": "^4.1.7",
 		"@types/whatwg-mimetype": "3.0.0",
-		"type-fest": "^3.8.0"
+		"bcp-47": "1",
+		"css-tree": "^2.3.1",
+		"debug": "^4.3.4",
+		"leven": "3",
+		"type-fest": "^3.8.0",
+		"whatwg-mimetype": "^3.0.0"
 	}
 }

--- a/packages/@markuplint/vue-spec/package.json
+++ b/packages/@markuplint/vue-spec/package.json
@@ -15,7 +15,7 @@
 		"build": "tsc",
 		"clean": "tsc --build --clean"
 	},
-	"devDependencies": {
+	"dependencies": {
 		"@markuplint/ml-spec": "3.7.0"
 	},
 	"peerDependencies": {

--- a/packages/markuplint/package.json
+++ b/packages/markuplint/package.json
@@ -19,13 +19,6 @@
 		"dev": "tsc --build --watch",
 		"clean": "tsc --build --clean"
 	},
-	"devDependencies": {
-		"@types/cli-color": "^2.0.2",
-		"@types/debug": "^4.1.7",
-		"@types/has-yarn": "^1.0.1",
-		"@types/meow": "^6.0.0",
-		"@types/uuid": "^9.0.0"
-	},
 	"dependencies": {
 		"@markuplint/create-rule-helper": "3.8.0",
 		"@markuplint/file-resolver": "3.8.0",
@@ -38,6 +31,11 @@
 		"@markuplint/ml-spec": "3.7.0",
 		"@markuplint/rules": "3.8.0",
 		"@markuplint/shared": "3.5.0",
+		"@types/cli-color": "^2.0.2",
+		"@types/debug": "^4.1.7",
+		"@types/has-yarn": "^1.0.1",
+		"@types/meow": "^6.0.0",
+		"@types/uuid": "^9.0.0",
 		"chokidar": "^3.5.3",
 		"cli-color": "^2.0.3",
 		"debug": "^4.3.4",

